src/reducers/errors.spec.tsx

```diff
@@ -0,0 +1,63 @@
+import reducer, { actions, initialState } from './errors';
+
+describe(__filename, () => {
+  describe('reducer', () => {
+    it('stores an error to show', () => {
+      const message = 'oops, I am an error';
+      const error = new Error(message);
+
+      const state = reducer(undefined, actions.showError({ error }));
+
+      expect(state).toMatchObject({
+        errors: [{ id: initialState.nextErrorId, message }],
+      });
+    });
+
+    it('increments the next error ID when storing a new error', () => {
+      let state = reducer(
+        undefined,
+        actions.showError({ error: new Error('Bad Request') }),
+      );
+      expect(state).toMatchObject({
+        nextErrorId: initialState.nextErrorId + 1,
+      });
+
+      state = reducer(
+        state,
+        actions.showError({ error: new Error('Bad Request, again') }),
+      );
+      expect(state).toMatchObject({
+        nextErrorId: initialState.nextErrorId + 2,
+      });
+    });
+
+    it('removes an error when dismissed', () => {
```

Comment:
Would it be worth having a test where there are multiple errors in state and we remove one that isn't the last one?

Reply:
yep, that's a good idea